lnrpc/routerrpc/router_server.go

```diff
@@ -487,6 +492,156 @@ func toRPCPairData(data *routing.TimedPairResult) *PairData {
 	return &rpcData
 }
 
+// ImportMissionControl imports the state provided to our internal mission
+// control. Only entries that are fresher than our existing state will be used.
+func (s *Server) ImportMissionControl(ctx context.Context,
+	req *ImportMissionControlRequest) (*ImportMissionControlResponse,
+	error) {
+
+	if len(req.Pairs) == 0 {
+		return nil, errors.New("at least one pair required for import")
+	}
+
+	snapshot := &routing.MissionControlSnapshot{
+		Pairs: make(
+			[]routing.MissionControlPairSnapshot, len(req.Pairs),
+		),
+	}
+
+	for i, pairResult := range req.Pairs {
+		pairSnapshot, err := toPairSnapshot(pairResult)
+		if err != nil {
+			return nil, err
+		}
+
+		snapshot.Pairs[i] = *pairSnapshot
+	}
+
+	err := s.cfg.RouterBackend.MissionControl.ImportHistory(snapshot)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ImportMissionControlResponse{}, nil
+}
+
+func toPairSnapshot(pairResult *PairHistory) (*routing.MissionControlPairSnapshot,
+	error) {
+
+	from, err := route.NewVertexFromBytes(pairResult.NodeFrom)
+	if err != nil {
+		return nil, err
+	}
+
+	to, err := route.NewVertexFromBytes(pairResult.NodeTo)
+	if err != nil {
+		return nil, err
+	}
+
+	if bytes.Equal(from[:], to[:]) {
+		return nil, errors.New("peer pubkeys must differ")
```

Comment:
nit: wdyt about rephrasing the error to something like "node pair source and destination must be different"? Maybe we could even refer to the pubkey itself (like below).

Reply:
nice idea! added prefix with from/to to all errors